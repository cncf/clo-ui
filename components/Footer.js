module.exports = require('../dist/cjs/components/Footer.js');
