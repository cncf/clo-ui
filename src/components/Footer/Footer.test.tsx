import { render, screen } from '@testing-library/react';

import { Footer } from './Footer';

const defaultProps = {
  logo: <>Logo</>,
  children: <span>Content</span>,
};

describe('Footer', () => {
  it('creates snapshot', () => {
    const { asFragment } = render(<Footer {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<Footer {...defaultProps} />);

    expect(screen.getByText('Logo')).toBeInTheDocument();
    expect(screen.getByText('Content')).toBeInTheDocument();
  });
});
