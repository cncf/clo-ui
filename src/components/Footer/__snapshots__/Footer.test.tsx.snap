// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer creates snapshot 1`] = `
<DocumentFragment>
  <footer
    class="py-5 footer undefined"
  >
    <div
      class="container-lg"
    >
      <div
        class="d-flex flex-row flex-wrap align-items-stretch justify-content-between text-light"
      >
        <span>
          Content
        </span>
        <div
          class="ms-0 ms-lg-auto fullMobileSection"
        >
          Logo
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
